google-http-client/src/main/java/com/google/api/client/http/HttpResponse.java

```diff
@@ -432,8 +432,10 @@ public void ignore() throws IOException {
    * @since 1.4
    */
   public void disconnect() throws IOException {
-    ignore();
+    // Close the connection before trying to close the InputStream content. If you are trying to
+    // disconnect, we shouldn't need to try to read any further content.
     response.disconnect();
+    ignore();
```

Comment:
I'm not sold this is the best solution -- the tests to pass however.

An alternative is to mark that this response is "disconnecting" and in `getContent()` don't read from the underlying response if we're "disconnecting".